### ISSUE TITLE
SYMCC custom mutator weird behaviour

### ISSUE THREAD

**Issue by Rumata888** [AFLplusplus/AFLplusplus#600]
It seems that when afl uses symcc custom mutator, at times when it queues in a new item from the corpus, it just freezes (until you hit enter).

**Comment by Rumata888** [comment]
This part seems very strange, to be honest (this is when the file is read from STDIN):
`close(0);
ck_write(0, data->mutator_buf, r, fn);`
And also, doesn't SYMCC require an EoF? Will it get it this way?

**Comment by Rumata888** [comment]
Ok, I managed to fix it. The stdin proxy was unusable, it has to be implemented differently. Will clean up, check that it works in both cases and submit a pull request.

**Comment by vanhauser-thc** [comment]
@Rumata888 sorry I have been super busy :(
I used it with a target that needs the input file name, so I did not test it for stdin.
Thanks for a PR fixing this!

**Comment by Rumata888** [comment]
Hm. Yes, I'm still having some problems with it. While I managed to activate symcc and feed it input alright (the queue stage), some weirdness happens later on. For some reason AFL decided to start symcc binary instead of one build for afl, which freezes the process. I have no idea, why that is happenning. But this only happens if I return data that was read from symcc-created files. If  I just return a constant string from afl_custom_fuzz, then everything is OK.

**Comment by Rumata888** [comment]
Ok.  I got to the root of the problem. When files are added to the queue with **add_to_queue**, where the call to custom mutator takes place, the files themselves are not yet created, unless they are from the initial corpus. So the file mode shouldn't have worked, too, except for initial corpus processing. And since your application compiled with symcc just detected that there was no file, it probably graciously exited without exhibiting the kind of problems that I got with stdin.
